src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorDiagnosticFactory.cs

```diff
@@ -178,15 +178,61 @@ public static RazorDiagnostic Create_DisallowedScriptTag(SourceSpan? source)
             return diagnostic;
         }
 
-        public static readonly RazorDiagnosticDescriptor Template_InvalidLocation =
+        public static readonly RazorDiagnosticDescriptor TemplateInvalidLocation =
             new RazorDiagnosticDescriptor(
             "BL9994",
             () => "Razor templates cannot be used in attributes.",
             RazorDiagnosticSeverity.Error);
 
-        public static RazorDiagnostic CreateTemplate_InvalidLocation(SourceSpan? source)
+        public static RazorDiagnostic Create_TemplateInvalidLocation(SourceSpan? source)
         {
-            return RazorDiagnostic.Create(Template_InvalidLocation, source ?? SourceSpan.Undefined);
+            return RazorDiagnostic.Create(TemplateInvalidLocation, source ?? SourceSpan.Undefined);
+        }
+
+        public static readonly RazorDiagnosticDescriptor ChildContentSetByAttributeAndBody =
+            new RazorDiagnosticDescriptor(
+                "BL9995",
+                () => "The child content property '{0}' is set by both the attribute and the element contents.",
+                RazorDiagnosticSeverity.Error);
+
+        public static RazorDiagnostic Create_ChildContentSetByAttributeAndBody(SourceSpan? source, string attribute)
+        {
+            return RazorDiagnostic.Create(ChildContentSetByAttributeAndBody, source ?? SourceSpan.Undefined, attribute);
+        }
+
+        public static readonly RazorDiagnosticDescriptor ChildContentMixedWithExplicitChildContent =
+            new RazorDiagnosticDescriptor(
+                "BL9996",
+                () => "Unrecognized child content. When using explicit child content, all significant content must be inside a " +
+                "child content element such as '<ChildContent>'. Check that the spelling of all child elements matches the component's" +
+                "declared parameters.",
+                RazorDiagnosticSeverity.Error);
```

Comment:
Not absolutely essential, but I wonder if we could make this easier to understand. Possible phrasing (if we have the necessary info):

```
Unexpected element 'div' inside component 'SuperTable'. The component 'SuperTable' accepts the following top-level items only: 'Header', 'Body', 'Footer'.
```

Is that possible, and does that match the intent of the diagnostic?

Reply:
Yeah, I agree that would be nicer. I'll see what it would take.

Reply:
I want to omit the offending text in the error message (`div` in your example), because we're going to add one error for the whole lot of content, and printing the text that appeared could be quite large in many cases. 

It's fairly straightforward to include the supported child content parameter names.